core/src/main/java/org/apache/iceberg/UpdateRequirements.java

```diff
@@ -173,6 +175,26 @@ private void update(MetadataUpdate.SetDefaultSortOrder unused) {
       }
     }
 
+    private void update(MetadataUpdate.RemovePartitionSpecs unused) {
+      // require that the default partition spec has not changed
+      if (!setSpecId) {
+        if (base != null && !isReplace) {
+          require(new UpdateRequirement.AssertDefaultSpecID(base.defaultSpecId()));
+        }
+        this.setSpecId = true;
+      }
+      // require that all the branch has not changed, so that old specs won't be written.
```

Comment:
```suggestion
      // require that the branch has not changed, so that old specs won't be written.
```

Reply:
I'd update to say `require that no branches have changed`